numpy/ma/core.py

```diff
@@ -786,22 +786,17 @@ def fix_invalid(a, mask=nomask, copy=True, fill_value=None):
 
 class _DomainCheckInterval(object):
     """
-    Define a valid interval, so that :
-
-    ``domain_check_interval(a,b)(x) == True`` where
-    ``x < a`` or ``x > b``.
+    Define a valid interval, so that:
 
+    ``_DomainCheckInterval(a,b)(x)`` is False where ``a <= x <= b``.
```

Comment:
Actually, if we think of using DomainCheckInterval even more generally - `a`, `x` and `b` don't even need to be numbers, so adding `or isnan(x)` might not be enough either. I think they can be any objects that can go through numpy ufuncs.

I think in practice none of this matters since the only "user" of DomainCheckInterval is the MaskedArray code which presumably masks Nans, and only calls it for numeric types. Nevertheless I don't think it's a good idea to make the comment less technically accurate than it was before.